### PR TITLE
Update Search Application API docs to discuss warnings

### PR DESCRIPTION
**Pull request by kderusso**
Updates Search Application API documentation to call out warnings returned in response headers and how to mitigate them.

### CODE REVIEW

File: docs/reference/search-application/apis/get-search-application.asciidoc [hunk 1]
```diff
@@ -8,7 +8,10 @@ beta::[]
 <titleabbrev>Get Search Application</titleabbrev>
 ++++
 
-Retrieves information about a Search Application.
+Retrieves information about a search application.
+
+If the search engine has an inconsistent state between its alias and configured indices, a warning header will be returned with the response.
```
Comment (leemthompo) [inline docs/reference/search-application/apis/get-search-application.asciidoc:13]:
```suggestion
If the search application has an inconsistent state between its alias and configured indices, a warning header will be returned with the response.
```

File: docs/reference/search-application/apis/search-application-search.asciidoc [hunk 2]
```diff
@@ -11,6 +11,9 @@ beta::[]
 Given specified query parameters, creates an Elasticsearch query to run. Any unspecified template parameters will be
 assigned their default values if applicable.
 
+If the search engine has an inconsistent state between its alias and configured indices, a warning header will be returned with the response.
```
Comment (leemthompo) [inline docs/reference/search-application/apis/search-application-search.asciidoc:14]:
```suggestion
If the search application has an inconsistent state between its alias and configured indices, a warning header will be returned with the response.
```

**Review by leemthompo — APPROVED** [review]
LGTM modulo typos :)

File: docs/reference/search-application/apis/get-search-application.asciidoc (hunk 1 above)
Comment (kderusso) [reply docs/reference/search-application/apis/get-search-application.asciidoc:13]: Thanks for the catch - I thought I was so good about not using engine anymore 😅

**Pull request merged**